modules/renter/hostdb/scan.go

```diff
@@ -248,18 +248,33 @@ func (hdb *HostDB) threadedScan() {
 		// pushing them further back in the hierarchy, ensuring that for the
 		// most part only online hosts are getting scanned unless there are
 		// fewer than hostCheckupQuantity of them.
-		//
-		// TODO: Cannot use SelectRandom (despite it being faster) because
-		// SelectRandom only returns active/online hosts. Need to be scanning
-		// the offline ones as well, otherwise a little bit of downtime is a
-		// death sentence for a host.
+
+		// Grab a set of hosts to scan, grab hosts that are active, inactive,
+		// and offline to get high diversity.
+		var onlineHosts, offlineHosts []modules.HostDBEntry
+		hosts := hdb.hostTree.All()
+		for i := 0; i < len(hosts) && (len(onlineHosts) < hostCheckupQuantity || len(offlineHosts) < hostCheckupQuantity); i++ {
+			// Figure out if the host is online or offline.
+			online := false
+			scanLen := len(hosts[i].ScanHistory)
+			if scanLen > 0 && hosts[i].ScanHistory[scanLen-1].Success {
+				online = true
+			}
```

Comment:
also, if you pull the second conditional out of the `for` statement, you can rewrite this as:
```go
for _, host := range hdb.hostTree.All() {
    online := len(host.ScanHistory) > 0 && host.ScanHistory[len(host.ScanHistory)-1].Success
    // ...

    // break once we have enough hosts
    if len(onlineHosts) >= hostCheckupQuantity || len(offlineHosts) >= hostCheckupQuantity {
        break
    }
}
```